src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/StackExchangeRedis2Plus/SessionCache.cs

```diff
@@ -71,6 +75,28 @@ public void Harvest(ISegment hostSegment)
             }
         }
 
+        private void CleanUp()
+        {
+            var cleanedSessions = 0;
+            foreach (var pair in _sessionCache)
```

Comment:
To follow up on Chris' above comment, I think we risk this enumerable changing due to `Harvest` as we loop through it.

Reply:
My mistake, this enumerable is threadsafe.